### DISCUSSION TITLE
how to insert batch

### DISCUSSION THREAD

**Discussion by guotie** [prisma/prisma#4809] [Q&A] [answered]
how to insert batch?

**Comment by nikolasburk** [comment, marked as answer]
Hey @guotie, thanks a lot for raising this! Prisma Client currently doesn't have a `createMany` operation, there are open issues for this [here](https://github.com/prisma/prisma-client-js/issues/835) and [here](https://github.com/prisma/prisma/issues/4349).

One workaround for this is to use the [`$transaction`](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#the-transaction-api) method and pass several `create` promises to it:

```ts
const userData: Prisma.UserCreateInput[] = [{
  name: 'James',
  email: 'james@prisma.io'
}, {
  name: 'Sarah',
  email: 'sarah@prisma.io'
}]

const createCalls: Promise<User>[] = []
for (let u of userData) {
  const c = prisma.user.create({ data: u })
  createCalls.push(c)
}
await prisma.$transaction(createCalls)
```

Let me know if that helps or if you have any further questions, always happy to help 🙌

**Comment by guotie** [comment]
thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
